### PR TITLE
gateio: update FuturesAccount and Position struct

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

* Updates struct to include all fields and converts string fields that hold a float to types.Number for easier conversion. 
* Fix bug that stopped transfers from anything outside of spot.
* Exported GateioTime -> Time so that it can be instantiable outside of package 
* Addresses issue #1544 

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shazbert** [comment]
Got an update to the fee struct coming through. PAUSE. 🙏

**Comment by shazbert** [comment]
It is now pushed you can all relax and get back to reviewing this massive PR. 👍

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Looks like there are a few more structs that could be changed to use Time. `WsOrderbookTickerData.UpdateTimeMS`, `WsCandlesticks.Timestamp`, `WsResponse.Time` etc etc etc but otherwise this is fine 🔥 🐶

**Review by thrasher- — APPROVED** [review]
Once the remaining time vars which @gloriousCode mentioned are resolved, this looks good to go!

File: exchanges/gateio/gateio_convert.go [hunk 1]
```diff
@@ -34,20 +35,20 @@ func (a *gateioTime) UnmarshalJSON(data []byte) error {
...
 		}
 		if math.Trunc(parsedValue) != parsedValue {
-			*a = gateioTime(time.UnixMicro(int64(parsedValue * 1e3))) // Account for "1691122380942.173000" microseconds
+			*a = Time(time.UnixMicro(int64(parsedValue * 1e3))) // Account for "1691122380942.173000" microseconds
 			return nil
 		}
 		standard = int64(parsedValue)
 	default:
-		return fmt.Errorf("cannot unmarshal %T into gateioTime", val)
+		return fmt.Errorf("cannot unmarshal %T into GateioTime", val)
```
Comment (thrasher-) [inline exchanges/gateio/gateio_convert.go:43]:
```suggestion
		return fmt.Errorf("cannot unmarshal %T into Time", val)
```

**Review by thrasher- — APPROVED** [review]
Changes ACK, thanks!

**Review by gloriousCode — APPROVED** [review]
Shazbert, thank you for taking the `Time` for these fixes. tACK

**Pull request merged**